Sources/NIOTCPEchoClient/Client.swift

```diff
@@ -0,0 +1,91 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#if swift(>=5.9)
+@_spi(AsyncChannel) import NIOCore
+@_spi(AsyncChannel) import NIOPosix
+
+@available(macOS 14, *)
+@main
+struct Client {
+    /// The host to connect to.
+    private let host: String
+    /// The port to connect to.
+    private let port: Int
+    /// The client's event loop group.
+    private let eventLoopGroup: any EventLoopGroup
+
+    static func main() async throws {
+        // We are creating this event loop group at the top level and it will live until
+        // the process exits. This means we also don't have to shut it down.
+        //
+        // Note that we start this group with 1 thread. In general, most NIO programs
+        // should use 1 thread as a default unless they're planning to be a network
+        // proxy or something that is expecting to be dominated by packet parsing. Most
+        // servers aren't, and NIO is very fast, so 1 NIO thread is quite capable of
+        // saturating the average small to medium sized machine.
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+
+        let server = Client(
+            host: "localhost",
+            port: 8765,
+            eventLoopGroup: eventLoopGroup
+        )
+        try await server.run()
+
+        print("Done sending requests; exiting in 5 seconds")
+        try await Task.sleep(for: .seconds(5))
```

Comment:
why do we sleep?

Reply:
If a user runs this in Xcode he gets a couple of seconds to see the output. Otherwise it is just gone immediately.

Reply:
Oof, that's pretty annoying. Could you add a comment explaining that and that it's not actually necessary? Examples get copy-pasted a lot so we don't want users to actually do this!

Reply:
Yeah, can we not do that? Let's find a better way that demonstrates this without just delaying it here...

Reply:
I mean I am happy to just not do this.